apps/prairielearn/src/lib/question.js

```diff
@@ -498,15 +511,31 @@ module.exports = {
     variant,
     check_submission_id,
     question,
-    course,
+    variant_course,
     authn_user_id,
     overrideGradeRateCheck,
     callback
   ) {
     debug('_gradeVariant()');
-    let questionModule, courseIssues, data, submission, grading_job;
+    let questionModule, question_course, courseIssues, data, submission, grading_job;
     async.series(
       [
+        (callback) => {
+          if (question.course_id === variant_course.id) {
+            question_course = variant_course;
+            callback(null);
+          } else {
+            sqldb.queryOneRow(
+              sql.select_question_course,
+              { question_course_id: question.course_id },
+              (err, result) => {
+                if (ERR(err, callback)) return;
+                question_course = result.rows[0].course;
+                callback(null);
+              }
+            );
+          }
+        },
```

Comment:
Let's make this an `async` function.

Reply:
I thought about it. This is annoying to do because if I make it an async function because the calling function is not async, so I would need to go make a bunch of things async, and I don't want that sort of refactor to be part of this. 

If I make it an old school asynchronous function with a callback, then calling such a function is not pretty, but I guess that's still better than this because the code is only in one place. I can go ahead and do that.

Reply:
done

Reply:
It's actually pretty straightforward to mix async + callback functions when using `async.series` (or really any `async.*`) function: just make the individual function async and the rest will work as normal. We use this all over the codebase to ease the transition to async functions. Here's an example from `grader_host`:

https://github.com/PrairieLearn/PrairieLearn/blob/0d62b3d6c622a1886d1f9985a119d206f0538ba2/apps/grader-host/src/index.js#L40-L58

Reply:
done